nucliadb_node/src/shards/shard_writer.rs

```diff
@@ -153,7 +153,7 @@ impl ShardWriter {
     }
 
     #[measure(actor = "shard", metric = "new")]
-    pub fn new(metadata: Arc<ShardMetadata>) -> NodeResult<ShardWriter> {
+    pub fn new(metadata: Arc<ShardMetadata>, vector_config: VectorConfig) -> NodeResult<ShardWriter> {
```

Comment:
Idea (feel free to ignore)
I've always thought it's a bit weird how we create a `ShardMetadata` in the cache and then use it to create a shard. I think it'd be cleaner to create a shard with a custom struct and then, either return a tuple `(ShardWriter, ShardMetadata)` or let the caller get the metadata from the writer. This struct will include this new vector config too.

Reply:
Yeah, I saw this and I agree. It's also weird that we create the shard inside the WriterCache instead of creating it in the writer and just setting it in the case.

I think they are changes worth doing, but belong to a separate PR.